Sources/Core/Extensions/URLExtension.swift

```diff
@@ -19,6 +19,13 @@ extension URL {
     }
 
     private static func make(from string: String, relativeTo baseURL: URL) -> URL {
+        let range = NSRange(location: 0, length: string.utf16.count)
+        // Checks that url paths with relative paths like /../ are not sent to the API
+        // swiftlint:disable:next force_try
+        let regex = try! NSRegularExpression(pattern: "\\/\\.+\\/")
+        if regex.firstMatch(in: string, options: [], range: range) != nil {
+            fatalError("An invalid path parameter exists in \(string). Relative path parameters cannot be passed.")
+        }
```

Comment:
This might be another formulation to consider.
```suggestion
        for c in string.components(separatedBy: "/") where c == ".." {
            fatalError("An invalid path parameter exists in \(string). Relative path parameters cannot be passed.")
        }
```

Reply:
I think this would catch `..` at the beginning and end of the string, which the current regex wouldn't.

Reply:
Just trying to filter dots in the path params to avoid people passing in relative paths, so not too worried about the .. in the beginning. Going to update the regex to handle .. at the end. Also, regex makes it possible to search for any number of dots in between slashes.  Also, if we want to make a stricter sanitization in the future we would only have to update the regex.

Reply:
Do we want to search for any number of dots?

Reply:
Not necessary to search for any number of dots because it seems unlikely, but if any other case does arise we do throw an error. Also, using regex keeps it consistent across the SDKs.